modules/space/rendering/renderablefluxnodes.cpp

```diff
@@ -0,0 +1,1093 @@
+ ****************************************************************************************/
+#include <modules/space/rendering/renderablefluxnodes.h>
+
+#include <openspace/engine/globals.h>
+#include <openspace/engine/windowdelegate.h>
+#include <openspace/interaction/navigationhandler.h>
+#include <openspace/interaction/orbitalnavigator.h>
+#include <openspace/rendering/renderengine.h>
+#include <openspace/scene/scene.h>
+#include <openspace/util/timemanager.h>
+#include <openspace/util/updatestructures.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/logging/logmanager.h>
+// Test debugging tools more then logmanager
+#include <ghoul/logging/consolelog.h>
+#include <ghoul/logging/visualstudiooutputlog.h>
+#include <ghoul/filesystem/cachemanager.h>
+
+#include <ghoul/opengl/programobject.h>
+#include <ghoul/opengl/textureunit.h>
+#include <functional> 
+#include <fstream>
+#include <thread>
+#include <openspace/query/query.h>
+#include <sys/stat.h>
+#include <optional>
+
+namespace {
+    // log category
+    constexpr const char* _loggerCat = "renderableFluxNodes";
+
+    // GL variables for shaders, probably needed some of them atleast
+    constexpr const GLuint VaPosition   = 0;    // MUST CORRESPOND TO THE SHADER PROGRAM
+    constexpr const GLuint VaColor      = 1;    // MUST CORRESPOND TO THE SHADER PROGRAM
+    constexpr const GLuint VaFiltering  = 2;    // MUST CORRESPOND TO THE SHADER PROGRAM
+
+    constexpr int8_t CurrentCacheVersion = 2;
+    
+    //streamColor, nodeSize, nodeSizeLargerFlux, thresholdFlux, 
+    constexpr const std::array<const char*, 26> UniformNames = {
+        "streamColor", "nodeSize", "nodeSizeLargerFlux", "thresholdFlux", "colorMode",
+        "filterLower", "filterUpper", "scalingMode", "colorTableRange", "domainLimZ",
+        "nodeSkip", "nodeSkipDefault", "nodeSkipEarth", "nodeSkipMethod", 
+        "nodeSkipFluxThreshold", "nodeSkipRadiusThreshold", "fluxColorAlpha", 
+        "fluxColorAlphaIlluminance", "earthPos", "distanceThreshold", 
+        "enhanceMethod", "flowColor", "usingParticles", 
+        "particleSize", "particleSpacing", "particleSpeed"
+    };
+    constexpr const std::array<const char*, 14> UniformNames2 = {
+        "time", "flowColoring", "maxNodeDistanceSize", "usingCameraPerspective",
+        "drawCircles", "drawHollow", "useGaussian", "usingRadiusPerspective",
+        "perspectiveDistanceFactor", "maxNodeSize", "minNodeSize", "usingPulse",
+        "usingGaussianPulse", "pulsatingAlways"
+    };
+
+    // --------------------------------- Property Info -------------------------------- //
+    constexpr openspace::properties::Property::PropertyInfo GoesEnergyBinsInfo = {
+        "GoesEnergy",
+        "Goes Energy",
+        "Select which energy bin you want to show. Emin01 is values > 10 Mev,"
+        "Default is Emin03 where values > 100 Mev."
+    };
+    constexpr openspace::properties::Property::PropertyInfo ColorModeInfo = {
+        "colorMode",
```

Comment:
First letter would be capitalized (same for other property identifiers in this file)